### modules

In [5]:
from selenium import webdriver as wbd
from selenium.webdriver.common.by import By
import pandas as pd
# import datetime as dt
import time
import csv
from openpyxl import Workbook
from bs4 import BeautifulSoup
import re
import datetime 

In [2]:
# from selenium.webdriver.chrome.service import Service
# from webdriver_manager.chrome import ChromeDriverManager
# from selenium.webdriver.common.keys import Keys


# options = wbd.ChromeOptions()
# options.add_argument("headless")
# options.add_argument("disable-gpu")
# driver = wbd.Chrome(service=Service(ChromeDriverManager().install()))

# 1. 공휴일 추출

In [14]:
options = wbd.ChromeOptions()
options.add_argument("headless")
options.add_argument("disable-gpu")
driver = wbd.Chrome("./chromedriver.exe", options=options)

html = 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=%EB%84%A4%EC%9D%B4%EB%B2%84%EB%8B%AC%EB%A0%A5&oquery=%EB%84%A4%E3%85%A3%E3%85%87%EB%B2%84%EB%8B%AC%EB%A0%A5&tqi=i4EPmlprvmZsslbDSkdssssssU0-356245'
driver.get(html)

C:\Users\jjgus\AppData\Local\Temp\ipykernel_27292\4121251317.py:4: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = wbd.Chrome("./chromedriver.exe", options=options)


## 1.1 연월 추출

In [15]:
html = BeautifulSoup(driver.page_source, 'lxml')
txt = html.find_all()
html.find_all('span', class_='_title')[0].get_text()

'2023.06'

## 1.2. 날짜별 color 추출


[value_of_css_property](https://stackoverflow.com/questions/42114108/python-using-selenium-value-of-css-property-different-in-output-and-inspection)
- property의 여러가지 elements name으로 css 요소 추출
- 여기서는 글자 색깔을가져와야 하므로 name argument로 'color'

In [16]:
driver.find_element(By.CLASS_NAME, 'cld_day').value_of_css_property('color')

'rgba(231, 231, 231, 1)'

In [17]:
# index와 color

for i, v in enumerate(driver.find_elements(By.CLASS_NAME, 'cld_day')):
    print(i, v.value_of_css_property('color'))

0 rgba(231, 231, 231, 1)
1 rgba(231, 231, 231, 1)
2 rgba(231, 231, 231, 1)
3 rgba(231, 231, 231, 1)
4 rgba(36, 36, 36, 1)
5 rgba(36, 36, 36, 1)
6 rgba(36, 36, 36, 1)
7 rgba(252, 76, 78, 1)
8 rgba(36, 36, 36, 1)
9 rgba(252, 76, 78, 1)
10 rgba(36, 36, 36, 1)
11 rgba(36, 36, 36, 1)
12 rgba(36, 36, 36, 1)
13 rgba(36, 36, 36, 1)
14 rgba(252, 76, 78, 1)
15 rgba(36, 36, 36, 1)
16 rgba(36, 36, 36, 1)
17 rgba(36, 36, 36, 1)
18 rgba(36, 36, 36, 1)
19 rgba(36, 36, 36, 1)
20 rgba(36, 36, 36, 1)
21 rgba(252, 76, 78, 1)
22 rgba(36, 36, 36, 1)
23 rgba(36, 36, 36, 1)
24 rgba(36, 36, 36, 1)
25 rgba(36, 36, 36, 1)
26 rgba(36, 36, 36, 1)
27 rgba(36, 36, 36, 1)
28 rgba(252, 76, 78, 1)
29 rgba(36, 36, 36, 1)
30 rgba(36, 36, 36, 1)
31 rgba(36, 36, 36, 1)
32 rgba(36, 36, 36, 1)
33 rgba(36, 36, 36, 1)
34 rgba(231, 231, 231, 1)


In [18]:
for i, v in enumerate(driver.find_elements(By.CLASS_NAME, 'cld_day')):
    print(i, v.value_of_css_property('color'))
    print(re.sub(r'[^0-9]', '', v.value_of_css_property('color')))

0 rgba(231, 231, 231, 1)
2312312311
1 rgba(231, 231, 231, 1)
2312312311
2 rgba(231, 231, 231, 1)
2312312311
3 rgba(231, 231, 231, 1)
2312312311
4 rgba(36, 36, 36, 1)
3636361
5 rgba(36, 36, 36, 1)
3636361
6 rgba(36, 36, 36, 1)
3636361
7 rgba(252, 76, 78, 1)
25276781
8 rgba(36, 36, 36, 1)
3636361
9 rgba(252, 76, 78, 1)
25276781
10 rgba(36, 36, 36, 1)
3636361
11 rgba(36, 36, 36, 1)
3636361
12 rgba(36, 36, 36, 1)
3636361
13 rgba(36, 36, 36, 1)
3636361
14 rgba(252, 76, 78, 1)
25276781
15 rgba(36, 36, 36, 1)
3636361
16 rgba(36, 36, 36, 1)
3636361
17 rgba(36, 36, 36, 1)
3636361
18 rgba(36, 36, 36, 1)
3636361
19 rgba(36, 36, 36, 1)
3636361
20 rgba(36, 36, 36, 1)
3636361
21 rgba(252, 76, 78, 1)
25276781
22 rgba(36, 36, 36, 1)
3636361
23 rgba(36, 36, 36, 1)
3636361
24 rgba(36, 36, 36, 1)
3636361
25 rgba(36, 36, 36, 1)
3636361
26 rgba(36, 36, 36, 1)
3636361
27 rgba(36, 36, 36, 1)
3636361
28 rgba(252, 76, 78, 1)
25276781
29 rgba(36, 36, 36, 1)
3636361
30 rgba(36, 36, 36, 1)
3636361
31 rgba(36, 36,

In [19]:
# 화면상 달력에 나타나는 실제 day sequence

html = BeautifulSoup(driver.page_source, 'lxml')
txt = html.find_all(class_ = 'cld_day')

for i in txt:
    print(i.string)

28
29
30
31
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
1


### ex1)

In [20]:
for i, v in enumerate(zip(driver.find_elements(By.CLASS_NAME, 'cld_day'), txt)):
    print(f'''{i}, {v[0].value_of_css_property('color')},{re.sub(r'[^0-9]', '', v[0].value_of_css_property('color'))} ,{v[1].string}''')

0, rgba(231, 231, 231, 1),2312312311 ,28
1, rgba(231, 231, 231, 1),2312312311 ,29
2, rgba(231, 231, 231, 1),2312312311 ,30
3, rgba(231, 231, 231, 1),2312312311 ,31
4, rgba(36, 36, 36, 1),3636361 ,1
5, rgba(36, 36, 36, 1),3636361 ,2
6, rgba(36, 36, 36, 1),3636361 ,3
7, rgba(252, 76, 78, 1),25276781 ,4
8, rgba(36, 36, 36, 1),3636361 ,5
9, rgba(252, 76, 78, 1),25276781 ,6
10, rgba(36, 36, 36, 1),3636361 ,7
11, rgba(36, 36, 36, 1),3636361 ,8
12, rgba(36, 36, 36, 1),3636361 ,9
13, rgba(36, 36, 36, 1),3636361 ,10
14, rgba(252, 76, 78, 1),25276781 ,11
15, rgba(36, 36, 36, 1),3636361 ,12
16, rgba(36, 36, 36, 1),3636361 ,13
17, rgba(36, 36, 36, 1),3636361 ,14
18, rgba(36, 36, 36, 1),3636361 ,15
19, rgba(36, 36, 36, 1),3636361 ,16
20, rgba(36, 36, 36, 1),3636361 ,17
21, rgba(252, 76, 78, 1),25276781 ,18
22, rgba(36, 36, 36, 1),3636361 ,19
23, rgba(36, 36, 36, 1),3636361 ,20
24, rgba(36, 36, 36, 1),3636361 ,21
25, rgba(36, 36, 36, 1),3636361 ,22
26, rgba(36, 36, 36, 1),3636361 ,23
27, rgba(36, 36

- 2312312311: grey -> drop
- 25276781: red -> 비영업일
- 3636361: black -> 영업일

### 뒤로가기하며 추출

In [23]:

# driver.get(html)

html = BeautifulSoup(driver.page_source, 'lxml')
txt = html.find_all(class_ = 'cld_day')
print(html.find_all('span', class_='_title')[0].get_text())

for i, v in enumerate(zip(driver.find_elements(By.CLASS_NAME, 'cld_day'), txt)):
    print({i}, {re.sub(r'[^0-9]', '', v[0].value_of_css_property('color'))}, {v[1].string})
    
driver.find_element(By.XPATH, value = '//*[@id="direct_calendar"]/div/div[1]/div[1]/a[2]').click()

2023.04
{0} {'2312312311'} {'26'}
{1} {'2312312311'} {'27'}
{2} {'2312312311'} {'28'}
{3} {'2312312311'} {'29'}
{4} {'2312312311'} {'30'}
{5} {'2312312311'} {'31'}
{6} {'3636361'} {'1'}
{7} {'25276781'} {'2'}
{8} {'3636361'} {'3'}
{9} {'3636361'} {'4'}
{10} {'3636361'} {'5'}
{11} {'3636361'} {'6'}
{12} {'3636361'} {'7'}
{13} {'3636361'} {'8'}
{14} {'25276781'} {'9'}
{15} {'3636361'} {'10'}
{16} {'3636361'} {'11'}
{17} {'3636361'} {'12'}
{18} {'3636361'} {'13'}
{19} {'3636361'} {'14'}
{20} {'3636361'} {'15'}
{21} {'25276781'} {'16'}
{22} {'3636361'} {'17'}
{23} {'3636361'} {'18'}
{24} {'3636361'} {'19'}
{25} {'3636361'} {'20'}
{26} {'3636361'} {'21'}
{27} {'3636361'} {'22'}
{28} {'25276781'} {'23'}
{29} {'3636361'} {'24'}
{30} {'3636361'} {'25'}
{31} {'3636361'} {'26'}
{32} {'3636361'} {'27'}
{33} {'3636361'} {'28'}
{34} {'3636361'} {'29'}
{35} {'25276781'} {'30'}
{36} {'2312312311'} {'1'}
{37} {'2312312311'} {'2'}
{38} {'2312312311'} {'3'}
{39} {'2312312311'} {'4'}
{40} {'2312312311'} 

252로시작하면 비영업일로 판단

## 1.3 반복분

In [24]:
# driver.get(html)

html = BeautifulSoup(driver.page_source, 'lxml')
txt = html.find_all(class_ = 'cld_day')
print(html.find_all('span', class_='_title')[0].get_text())

for i, v in enumerate(zip(driver.find_elements(By.CLASS_NAME, 'cld_day'), txt)):
    print(f'''idx: {i}, color: {re.sub(r'[^0-9]', '', v[0].value_of_css_property('color'))}, date: {html.find_all('span', class_='_title')[0].get_text()}.{v[1].string}''')
    
driver.find_element(By.XPATH, value = '//*[@id="direct_calendar"]/div/div[1]/div[1]/a[2]').click()

2023.03
idx: 0, color: 2312312311, date: 2023.03.26
idx: 1, color: 2312312311, date: 2023.03.27
idx: 2, color: 2312312311, date: 2023.03.28
idx: 3, color: 25276781, date: 2023.03.1
idx: 4, color: 3636361, date: 2023.03.2
idx: 5, color: 3636361, date: 2023.03.3
idx: 6, color: 3636361, date: 2023.03.4
idx: 7, color: 25276781, date: 2023.03.5
idx: 8, color: 3636361, date: 2023.03.6
idx: 9, color: 3636361, date: 2023.03.7
idx: 10, color: 3636361, date: 2023.03.8
idx: 11, color: 3636361, date: 2023.03.9
idx: 12, color: 3636361, date: 2023.03.10
idx: 13, color: 3636361, date: 2023.03.11
idx: 14, color: 25276781, date: 2023.03.12
idx: 15, color: 3636361, date: 2023.03.13
idx: 16, color: 3636361, date: 2023.03.14
idx: 17, color: 3636361, date: 2023.03.15
idx: 18, color: 3636361, date: 2023.03.16
idx: 19, color: 3636361, date: 2023.03.17
idx: 20, color: 3636361, date: 2023.03.18
idx: 21, color: 25276781, date: 2023.03.19
idx: 22, color: 3636361, date: 2023.03.20
idx: 23, color: 3636361, date: 2

In [25]:

datetime.datetime(2020, 4, 2)

datetime.datetime(2020, 4, 2, 0, 0)

### 1달 case

In [67]:
# driver.get(html)

html = BeautifulSoup(driver.page_source, 'lxml')
txt = html.find_all(class_ = 'cld_day')
yyyymm = html.find_all('span', class_='_title')[0].get_text()
print(f'[{yyyymm}]')



for i, v in enumerate(zip(driver.find_elements(By.CLASS_NAME, 'cld_day'), txt)):
    
    tmp_list = []
    
    color = re.sub(r'[^0-9]', '', v[0].value_of_css_property('color'))
    day = v[1].string
    
    if color == '2312312311' or color == '3636361': # 회색, 검은색은 skip
        continue
    
    print(f'''idx: {i}, color: {color}, day: {day}''')
    year = int(yyyymm[:4])
    month = int(yyyymm[5:7].lstrip("0"))
    day = int(day.lstrip("0"))
    print(year, month, day)
    print(datetime.datetime(year, month, day))
    # tmp_list.append(datetime.datetime(yyyymm[:3], yyyymm[4:6], day))
    
# tmp_dict[yyyymm] = []
    
# 한달 전으로 이동
driver.find_element(By.XPATH, value = '//*[@id="direct_calendar"]/div/div[1]/div[1]/a[2]').click()

[2023.04]
idx: 7, color: 25276781, day: 2
2023 4 2
2023-04-02 00:00:00
idx: 14, color: 25276781, day: 9
2023 4 9
2023-04-09 00:00:00
idx: 21, color: 25276781, day: 16
2023 4 16
2023-04-16 00:00:00
idx: 28, color: 25276781, day: 23
2023 4 23
2023-04-23 00:00:00
idx: 35, color: 25276781, day: 30
2023 4 30
2023-04-30 00:00:00


### 여러달 case

In [13]:
options = wbd.ChromeOptions()
options.add_argument("headless")
options.add_argument("disable-gpu")
driver = wbd.Chrome("./chromedriver.exe", options=options)

html = 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=%EB%84%A4%EC%9D%B4%EB%B2%84%EB%8B%AC%EB%A0%A5&oquery=%EB%84%A4%E3%85%A3%E3%85%87%EB%B2%84%EB%8B%AC%EB%A0%A5&tqi=i4EPmlprvmZsslbDSkdssssssU0-356245'
driver.get(html)

C:\Users\jjgus\AppData\Local\Temp\ipykernel_22476\4121251317.py:4: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = wbd.Chrome("./chromedriver.exe", options=options)


In [25]:
options = wbd.ChromeOptions()
# options.add_argument("headless")
options.add_argument("disable-gpu")
driver = wbd.Chrome("./chromedriver.exe", options=options)

html = 'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=%EB%84%A4%EC%9D%B4%EB%B2%84%EB%8B%AC%EB%A0%A5&oquery=%EB%84%A4%E3%85%A3%E3%85%87%EB%B2%84%EB%8B%AC%EB%A0%A5&tqi=i4EPmlprvmZsslbDSkdssssssU0-356245'
driver.get(html)

C:\Users\jjgus\AppData\Local\Temp\ipykernel_22476\2057289555.py:4: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = wbd.Chrome("./chromedriver.exe", options=options)


In [26]:
nonBusiness_list = []


for _ in range(24):
    # driver.get(html)

    html = BeautifulSoup(driver.page_source, 'lxml')
    txt = html.find_all(class_ = 'cld_day')
    yyyymm = html.find_all('span', class_='_title')[0].get_text()
    print(f'\n[{yyyymm}]')



    for i, v in enumerate(zip(driver.find_elements(By.CLASS_NAME, 'cld_day'), txt)):

        tmp_list = []

        color = re.sub(r'[^0-9]', '', v[0].value_of_css_property('color'))
        day = v[1].string

        if color == '2312312311' or color == '3636361': # 회색, 검은색은 skip
            continue

        print(f'''idx: {i}, color: {color}, day: {day}''')
        year = int(yyyymm[:4])
        month = int(yyyymm[5:7].lstrip("0"))
        day = int(day.lstrip("0"))
        # print(year, month, day)
        # print(datetime.datetime(year, month, day))
        nonBusiness_list.append(datetime.datetime(year, month, day))

    # 한달 전으로 이동
    driver.find_element(By.XPATH, value = '//*[@id="direct_calendar"]/div/div[1]/div[1]/a[2]').click()
    time.sleep(2)


[2023.06]
idx: 7, color: 25276781, day: 4
idx: 9, color: 25276781, day: 6
idx: 14, color: 25276781, day: 11
idx: 21, color: 25276781, day: 18
idx: 28, color: 25276781, day: 25

[2023.05]
idx: 5, color: 25276781, day: 5
idx: 7, color: 25276781, day: 7
idx: 14, color: 25276781, day: 14
idx: 21, color: 25276781, day: 21
idx: 27, color: 25276781, day: 27
idx: 28, color: 25276781, day: 28
idx: 29, color: 25276781, day: 29

[2023.04]
idx: 7, color: 25276781, day: 2
idx: 14, color: 25276781, day: 9
idx: 21, color: 25276781, day: 16
idx: 28, color: 25276781, day: 23
idx: 35, color: 25276781, day: 30

[2023.03]
idx: 3, color: 25276781, day: 1
idx: 7, color: 25276781, day: 5
idx: 14, color: 25276781, day: 12
idx: 21, color: 25276781, day: 19
idx: 28, color: 25276781, day: 26

[2023.02]
idx: 7, color: 25276781, day: 5
idx: 14, color: 25276781, day: 12
idx: 21, color: 25276781, day: 19
idx: 28, color: 25276781, day: 26

[2023.01]
idx: 0, color: 25276781, day: 1
idx: 7, color: 25276781, day: 8
idx

ElementClickInterceptedException: Message: element click intercepted: Element <a nocr="" href="#" title="이전" class="sp cld_pre _prev_month">...</a> is not clickable at point (448, 28). Other element would receive the click: <input type="text" id="nx_query" name="query" class="box_window" maxlength="255" accesskey="s" value="네이버달력" autocomplete="off" placeholder="검색어를 입력해 주세요." data-atcmp-element="">
  (Session info: chrome=114.0.5735.111)
Stacktrace:
Backtrace:
	GetHandleVerifier [0x00276E73+48323]
	(No symbol) [0x00209661]
	(No symbol) [0x00115308]
	(No symbol) [0x00145202]
	(No symbol) [0x00143D38]
	(No symbol) [0x0014253B]
	(No symbol) [0x0014193E]
	(No symbol) [0x0013A0FC]
	(No symbol) [0x0015A8DC]
	(No symbol) [0x00139BD6]
	(No symbol) [0x0015AC34]
	(No symbol) [0x0016CAC2]
	(No symbol) [0x0015A6D6]
	(No symbol) [0x0013847C]
	(No symbol) [0x0013957D]
	GetHandleVerifier [0x004DFD5D+2575277]
	GetHandleVerifier [0x0051F86E+2836158]
	GetHandleVerifier [0x005196DC+2811180]
	GetHandleVerifier [0x003041B0+626688]
	(No symbol) [0x0021314C]
	(No symbol) [0x0020F4B8]
	(No symbol) [0x0020F59B]
	(No symbol) [0x002021B7]
	BaseThreadInitThunk [0x76367D59+25]
	RtlInitializeExceptionChain [0x76F1B74B+107]
	RtlClearBits [0x76F1B6CF+191]
	(No symbol) [0x00000000]


In [ ]:
nonBusiness_list

In [18]:
nonBusiness_list[0].strftime("%y-%m-%d")

'23-06-04'

In [23]:
nonBusiness_list

[datetime.datetime(2023, 6, 4, 0, 0),
 datetime.datetime(2023, 6, 6, 0, 0),
 datetime.datetime(2023, 6, 11, 0, 0),
 datetime.datetime(2023, 6, 18, 0, 0),
 datetime.datetime(2023, 6, 25, 0, 0),
 datetime.datetime(2023, 5, 5, 0, 0),
 datetime.datetime(2023, 5, 7, 0, 0),
 datetime.datetime(2023, 5, 14, 0, 0),
 datetime.datetime(2023, 5, 21, 0, 0),
 datetime.datetime(2023, 5, 27, 0, 0),
 datetime.datetime(2023, 5, 28, 0, 0),
 datetime.datetime(2023, 5, 29, 0, 0),
 datetime.datetime(2023, 4, 2, 0, 0),
 datetime.datetime(2023, 4, 9, 0, 0),
 datetime.datetime(2023, 4, 16, 0, 0),
 datetime.datetime(2023, 4, 23, 0, 0),
 datetime.datetime(2023, 4, 30, 0, 0),
 datetime.datetime(2023, 3, 1, 0, 0),
 datetime.datetime(2023, 3, 5, 0, 0),
 datetime.datetime(2023, 3, 12, 0, 0),
 datetime.datetime(2023, 3, 19, 0, 0),
 datetime.datetime(2023, 3, 26, 0, 0),
 datetime.datetime(2023, 2, 5, 0, 0),
 datetime.datetime(2023, 2, 12, 0, 0),
 datetime.datetime(2023, 2, 19, 0, 0),
 datetime.datetime(2023, 2, 26, 0,

# 2. 주말 체크

In [38]:
#날짜 리스트 만들기
import pandas as pd
start_date='20220101'
end_date='20230617'
date_list=pd.date_range(start=start_date, end=end_date, freq='D')
print(date_list)

DatetimeIndex(['2022-01-01', '2022-01-02', '2022-01-03', '2022-01-04',
               '2022-01-05', '2022-01-06', '2022-01-07', '2022-01-08',
               '2022-01-09', '2022-01-10',
               ...
               '2023-06-08', '2023-06-09', '2023-06-10', '2023-06-11',
               '2023-06-12', '2023-06-13', '2023-06-14', '2023-06-15',
               '2023-06-16', '2023-06-17'],
              dtype='datetime64[ns]', length=533, freq='D')


In [46]:
for i in date_list:
    print(i.weekday())

5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0
1
2
3
4
5
6
0


In [52]:
weekend_list = []

for i in date_list:
    if i.weekday() > 4: 
        weekend = True  # 오늘이 공휴일이면 True
        print(i, weekend)
        weekend_list.append(i)

2022-01-01 00:00:00 True
2022-01-02 00:00:00 True
2022-01-08 00:00:00 True
2022-01-09 00:00:00 True
2022-01-15 00:00:00 True
2022-01-16 00:00:00 True
2022-01-22 00:00:00 True
2022-01-23 00:00:00 True
2022-01-29 00:00:00 True
2022-01-30 00:00:00 True
2022-02-05 00:00:00 True
2022-02-06 00:00:00 True
2022-02-12 00:00:00 True
2022-02-13 00:00:00 True
2022-02-19 00:00:00 True
2022-02-20 00:00:00 True
2022-02-26 00:00:00 True
2022-02-27 00:00:00 True
2022-03-05 00:00:00 True
2022-03-06 00:00:00 True
2022-03-12 00:00:00 True
2022-03-13 00:00:00 True
2022-03-19 00:00:00 True
2022-03-20 00:00:00 True
2022-03-26 00:00:00 True
2022-03-27 00:00:00 True
2022-04-02 00:00:00 True
2022-04-03 00:00:00 True
2022-04-09 00:00:00 True
2022-04-10 00:00:00 True
2022-04-16 00:00:00 True
2022-04-17 00:00:00 True
2022-04-23 00:00:00 True
2022-04-24 00:00:00 True
2022-04-30 00:00:00 True
2022-05-01 00:00:00 True
2022-05-07 00:00:00 True
2022-05-08 00:00:00 True
2022-05-14 00:00:00 True
2022-05-15 00:00:00 True


In [53]:
from datetime import datetime
from pytimekr import pytimekr

# 현재 시간을 가져온다.
now = datetime.now()
day = now.weekday()
today = now.today().strftime("%Y-%m-%d")

#리스트 형태로 반환
list = pytimekr.holidays() 

hoilday = False

for i in list:
    if(today == str(i)): 
        hoilday = True  # 오늘이 공휴일이면 True

if day > 4 or hoilday: # 오늘이 주말이거나 공휴일이면
    print("hoilday")
else:
    print("weekday")

hoilday


In [54]:
weekend_list

[Timestamp('2022-01-01 00:00:00', freq='D'),
 Timestamp('2022-01-02 00:00:00', freq='D'),
 Timestamp('2022-01-08 00:00:00', freq='D'),
 Timestamp('2022-01-09 00:00:00', freq='D'),
 Timestamp('2022-01-15 00:00:00', freq='D'),
 Timestamp('2022-01-16 00:00:00', freq='D'),
 Timestamp('2022-01-22 00:00:00', freq='D'),
 Timestamp('2022-01-23 00:00:00', freq='D'),
 Timestamp('2022-01-29 00:00:00', freq='D'),
 Timestamp('2022-01-30 00:00:00', freq='D'),
 Timestamp('2022-02-05 00:00:00', freq='D'),
 Timestamp('2022-02-06 00:00:00', freq='D'),
 Timestamp('2022-02-12 00:00:00', freq='D'),
 Timestamp('2022-02-13 00:00:00', freq='D'),
 Timestamp('2022-02-19 00:00:00', freq='D'),
 Timestamp('2022-02-20 00:00:00', freq='D'),
 Timestamp('2022-02-26 00:00:00', freq='D'),
 Timestamp('2022-02-27 00:00:00', freq='D'),
 Timestamp('2022-03-05 00:00:00', freq='D'),
 Timestamp('2022-03-06 00:00:00', freq='D'),
 Timestamp('2022-03-12 00:00:00', freq='D'),
 Timestamp('2022-03-13 00:00:00', freq='D'),
 Timestamp

In [28]:
pytimekr.holidays() 

[datetime.date(2023, 1, 21),
 datetime.date(2023, 1, 22),
 datetime.date(2023, 1, 23),
 datetime.date(2023, 9, 28),
 datetime.date(2023, 9, 29),
 datetime.date(2023, 9, 30),
 datetime.date(2023, 1, 1),
 datetime.date(2023, 3, 1),
 datetime.date(2023, 5, 5),
 datetime.date(2023, 5, 26),
 datetime.date(2023, 6, 6),
 datetime.date(2023, 8, 15),
 datetime.date(2023, 10, 9),
 datetime.date(2023, 10, 3),
 datetime.date(2023, 12, 25)]

# QuantLib